File: Semester2/OOP/Homework/Test1Preparation/src/Car.cpp
#include "../includes/Car.h"


Car::Car() : name(""), model(""), year(0), color("") {}

Car::Car(const std::string &name, const std::string &model, int year, const std::string &color) : name(name), model(model), year(year), color(color) {}

Car::Car(const Car &other) : name(other.getName()), model(other.getModel()), year(other.getYear()), color(other.getColor()) {}

std::string Car::getName() const {
    return name;
}

std::string Car::getModel() const {
    return model;
}

int Car::getYear() const {
    return year;
}

std::string Car::getColor() const {
    return color;
}

bool Car::operator==(const Car &other) const {
    return model == other.getModel() && year == other.getYear();
}

bool Car::operator!=(const Car &other) const {
    return !(*this == other);
}

Car &Car::operator=(const Car &other)
{
    if (this != &other) {
        name = other.getName();
        model = other.getModel();
        year = other.getYear();
        color = other.getColor();
    }
    return *this;
}

std::string Car::toString() const {
    return name + " " + model + " " + std::to_string(year) + " " + color; 
}

File: Semester3/MAP/a1/src/model/Car.java
package model;

public class Car extends AbstractVehicle {
    private int numberOfDoors;

    public Car() {
        super();
        numberOfDoors = 0;
    }

    public Car(String name, int repairPrice, int numberOfDoors) {
        super(name, repairPrice);
        this.numberOfDoors = numberOfDoors;
    }

    public int getNumberOfDoors() {
        return numberOfDoors;
    }

    public void setNumberOfDoors(int numberOfDoors) {
        this.numberOfDoors = numberOfDoors;
    }

    @Override
    public String toString() {
        return super.toString() + " " + numberOfDoors;
    }
}
